vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -185,13 +185,10 @@ fire_and_forget WinRTWebSocketResource::PerformPing() noexcept
   }
 }
 
-fire_and_forget WinRTWebSocketResource::PerformWrite() noexcept
+fire_and_forget WinRTWebSocketResource::PerformWrite(string&& message, bool isBinary) noexcept
```

Comment:
Definitely think it's cleaner to get rid of the double queues. Github won't let me comment on the spot, but I think we might need to change some more logic to accommodate this change. We add to the dispatch queue after resuming on the coroutine on the background thread + waiting on a signal. I.e. I think it's possible to lose ordering before we submit the task to the second queue. 

Looking more broadly, have you considered doing writes on the original thread? The Windows APIs are async already, so they'll have they're on queueing and background thread under the hood. Not sure how much risk you're wanting to take on with changes to this though since we already have prod usage.

Reply:
Makes sense.

We do have a test that attempts to verify message sequencing, but it could have passed by chance.

I'll try enqueuing can be done at the beginning of `PerformWrite`.

Reply:
> The Windows APIs are async already, so they'll have they're on queueing and background thread under the hood

Will they?
Isn't `resume_background()` needed to ensure the async work doesn't happen in the original thread?
In any case, the actual `SotreAsync` call is the last part of this method. There is buffer preparing and string/array copying happening before in the current thread. We should keep all of this outside of the native module thread.

Reply:
The windows APIs will do the networking on a background thread.  There is no need to move to a background thread before calling them.  

I'm not sure if the buffer preparing etc needs to happen on a background thread or not.  Its certainly trivial in comparison to the network calls, but I suppose it could still be substantial.  Having said that, if someone is really doing that much data over JS network calls they should probably be looking at some custom native modules to avoid all that bridge traffic, since it would be slow no matter what.

Reply:
@NickGerleman 

I have enhanced test `WebSocketIntegrationTest::SendConsecutive` to also ensure message sequencing.
It further proves your point:
```
 Message: 
    Assert failed. Expected:<ABCDEFGHIJ> Actual:<BAIHGFJDEC>
```

I'll apply the necessary changes, but we now have coverage to prevent sequencing regressions.